components/empty/simple.tsx

```diff
@@ -1,25 +1,20 @@
-import { TinyColor } from '@ctrl/tinycolor';
 import * as React from 'react';
-import { useMemo } from 'react';
+import { TinyColor } from '@ctrl/tinycolor';
+
 import { useToken } from '../theme/internal';
 
 const Simple: React.FC = () => {
   const [, token] = useToken();
 
-  const { colorFill, colorFillTertiary, colorFillQuaternary, colorBgContainer } = token;
-
-  const { borderColor, shadowColor, contentColor } = useMemo(
-    () => ({
-      borderColor: new TinyColor(colorFill).onBackground(colorBgContainer).toHexShortString(),
-      shadowColor: new TinyColor(colorFillTertiary)
-        .onBackground(colorBgContainer)
-        .toHexShortString(),
-      contentColor: new TinyColor(colorFillQuaternary)
-        .onBackground(colorBgContainer)
-        .toHexShortString(),
-    }),
-    [colorFill, colorFillTertiary, colorFillQuaternary, colorBgContainer],
-  );
+  const borderColor = new TinyColor(token.colorFill)
```

Comment:
It's not useless. For each re-render, the calculated color is same which should be cached.